test/k8s/egress.go

```diff
@@ -151,6 +150,27 @@ var _ = SkipDescribeIf(func() bool {
 		Expect(ctEntriesAfterConnection - ctEntriesBeforeConnection).Should(Equal(1))
 	}
 
+	testEgressGatewayExclusionCIDR := func(fromGateway bool) {
+		if fromGateway {
+			By("Check egress policy from gateway node")
+		} else {
+			By("Check egress policy from non-gateway node")
+		}
+
+		hostIP := k8s1IP
+		if fromGateway {
+			hostIP = k8s2IP
+		}
+
+		src, _ := fetchPodsWithOffset(kubectl, randomNamespace, "client", testDSClient, hostIP, false, 1)
```

Comment:
Nit: Couldn't we merge this with `testEgressGateway` above with a flag to decide what case we're checking?

Reply:
Pushed https://github.com/cilium/cilium/pull/23448/commits/113ea0e834cfb59c7951cd276d3365d26eb3b677 to ease the review (and test it), will squash it with the manager changes once we are happy